File: print_shop_desktop/lib/services/api_service.dart
import 'dart:convert';
import 'package:flutter/foundation.dart';
import 'package:http/http.dart' as http;
import '../models/order.dart';
import '../models/print_config.dart';

class ApiService {
  static const String baseUrl = 'http://192.168.0.100:8000';
  final http.Client _client = http.Client();

  Future<bool> testConnection() async {
    try {
      debugPrint('Testing connection to $baseUrl/health');
      final response = await _client.get(
        Uri.parse('$baseUrl/health'),
        headers: {'Accept': 'application/json'},
      ).timeout(const Duration(seconds: 5));
      return response.statusCode == 200;
    } catch (e) {
      debugPrint('Connection error: $e');
      return false;
    }
  }

  Future<List<Order>> getOrders() async {
    try {
      final response = await _client.get(Uri.parse('$baseUrl/orders'));
      if (response.statusCode == 200) {
        final List<dynamic> data = json.decode(response.body);
        return data.map((json) => Order.fromJson(json)).toList();
      }
      throw Exception('Failed to load orders');
    } catch (e) {
      debugPrint('Error getting orders: $e');
      return [];
    }
  }

  Future<bool> updateOrderStatus(String orderId, String status) async {
    final response = await _client.patch(
      Uri.parse('$baseUrl/orders/$orderId'),
      body: json.encode({'status': status}),
      headers: {'Content-Type': 'application/json'},
    );
    return response.statusCode == 200;
  }
}

File: print_shop_desktop/services/api_service.py
import requests
from typing import Dict, List
import logging

class ApiService:
    def __init__(self):
        self.base_url = "http://192.168.0.100:8000"
        self.session = requests.Session()
        self.logger = logging.getLogger(__name__)

    def test_connection(self) -> bool:
        try:
            response = self.session.get(f"{self.base_url}/health")
            return response.status_code == 200
        except Exception as e:
            self.logger.error(f"Connection error: {e}")
            return False

    def get_orders(self) -> List[Dict]:
        try:
            response = self.session.get(f"{self.base_url}/orders")
            if response.status_code == 200:
                return response.json()
            raise Exception(f"Failed to get orders: {response.text}")
        except Exception as e:
            self.logger.error(f"Error getting orders: {e}")
            return []

    def update_order_status(self, order_id: str, status: str) -> bool:
        try:
            response = self.session.patch(
                f"{self.base_url}/orders/{order_id}",
                json={"status": status}
            )
            return response.status_code == 200
        except Exception as e:
            self.logger.error(f"Error updating order status: {e}")
            return False